pkg/roll/run_update.go

```diff
@@ -238,10 +230,16 @@ func (u update) lockRCs() error {
 	}
 
 	err = u.lock.Lock(oldPath)
+	if err != nil {
+		// The second key couldn't be locked, so release the first key before retrying.
+		RetryOrQuit(
+			func() error { return u.lock.Unlock(newPath) },
+			done,
+			u.logger,
+			fmt.Sprintf("unlocking %s", newPath),
+		)
+	}
 	if _, ok := err.(kp.AlreadyLockedError); ok {
```

Comment:
If no lock was acquired in the first place, control flow shouldn't come here :)

But if you're asking, "Does `Unlock()` return an error if the lock isn't held?" the answer is maybe.  If the key is not held by any session, an attempt to unlock it will succeed.  If the lock is held by a different session, it's an error.  If control flow makes it here and the lock is held by a different sessions, the locker's session must have been lost.